web/web/api/v1/service_location.py

```diff
@@ -126,15 +121,12 @@ def add_service_location():
     new_service_location = request.get_json()
 
     try:
-        does_service_location_exist = ServiceLocation.query.filter_by(
-            service_location_id=new_service_location['service_location_id']
-        ).count() > 0
+        does_service_location_exist = ServiceLocation.query.filter_by(service_location_id=new_service_location['alternate_service_location_id']).count() > 0
 
         if does_service_location_exist:
-            raise IntegrityError('Service location already exists', None, None)
-
-        new_service_location = service_location_schema.load(
-            new_service_location, session=db.session)
+            raise IntegrityError('Service location with alternate service location number already exists', None, None)
```

Comment:
Why are we checking if a service location exists and the re-raising the exception it would've thrown anyways? Just try to insert into the DB and handle the exception as you are currently doing.